File: gnsals0904/level1/최대공약수와_최소공배수.java
class Solution {
    public int[] solution(int n, int m) {
        int gcd = gcd(n, m);
        int lcm = n / gcd * m;

        return new int[]{gcd, lcm};
    }

    private int gcd(int a, int b) {
        while (b != 0) {
            int temp = a % b;
            a = b;
            b = temp;
        }

        return a;
    }
}

File: yujin-level0/Week66/최대공약수와 최소공배수.js
function solution(n, m) {
  let a = n;
  let b = m;

  // 나머지가 0이 될 때까지 반복
  while (b !== 0) {
    let remainder = a % b;
    a = b;
    b = remainder;
  }

  const gcd = a;
  const lcm = (n * m) / gcd;

  return [gcd, lcm];
}

// 다른 풀이
// function gcdlcm(a, b) {
//     let r;
//     for(let ab= a*b;r = a % b;a = b, b = r){}
//     return [b, ab/b];
// }
